Inc/blackpill_map.h

```diff
@@ -0,0 +1,217 @@
+ #if Y_AUTO_SQUARE
+  #define Y2_LIMIT_PIN       10 // PORT GPIOB Doubling up on Z- Axis Limit Switch as they will never be used at the same time
+  #define Y2_LIMIT_BIT       (1<<Y2_LIMIT_PIN)
+ #endif
+#endif
+
+#define Z_STEP_PIN          4
+#define Z_STEP_BIT          (1<<Z_STEP_PIN)
+#define Z_DIRECTION_PIN     5
+#define Z_DIRECTION_BIT     (1<<Z_DIRECTION_PIN)
+
+#if Z_GANGED || Z_AUTO_SQUARE
+ #define Z2_STEP_PIN         9
+ #define Z2_STEP_BIT         (1<<Z2_STEP_PIN)
+ #define Z2_DIRECTION_PIN    10
+ #define Z2_DIRECTION_BIT    (1<<Z2_DIRECTION_PIN)
+ #if Z_AUTO_SQUARE
+  #define Z2_LIMIT_PIN       10 // PORT GPIOB
+  #define Z2_LIMIT_BIT       (1<<Z2_LIMIT_PIN)
+ #endif
+
+#endif
+
+#if N_AXIS > 3
+ #define A_STEP_PIN          6
+ #define A_STEP_BIT          (1<<A_STEP_PIN)
+ #define A_DIRECTION_PIN     7
+ #define A_DIRECTION_BIT     (1<<A_DIRECTION_PIN)
+
+ #if X_GANGED || X_AUTO_SQUARE
+  #define STEP_MASK           (X_STEP_BIT|X2_STEP_BIT|Y_STEP_BIT|Z_STEP_BIT|A_STEP_BIT) // All step bits
+  #define DIRECTION_MASK      (X_DIRECTION_BIT|X2_DIRECTION_BIT|Y_DIRECTION_BIT|Z_DIRECTION_BIT|A_DIRECTION_BIT) // All direction bits
+ #elif   Y_GANGED || Y_AUTO_SQUARE
+  #define STEP_MASK           (X_STEP_BIT|Y_STEP_BIT|Y2_STEP_BIT|Z_STEP_BIT|A_STEP_BIT) // All step bits
+  #define DIRECTION_MASK      (X_DIRECTION_BIT|Y_DIRECTION_BIT|Y2_DIRECTION_BIT|Z_DIRECTION_BIT|A_DIRECTION_BIT) // All direction bits
+ #elif   Z_GANGED || Z_AUTO_SQUARE
+  #define STEP_MASK           (X_STEP_BIT|Y_STEP_BIT|Z_STEP_BIT|Z2_STEP_BIT|A_STEP_BIT) // All step bits
+  #define DIRECTION_MASK      (X_DIRECTION_BIT|Y_DIRECTION_BIT|Z_DIRECTION_BIT|Z2_DIRECTION_BIT|A_DIRECTION_BIT) // All direction bits
+ #else
+  #define STEP_MASK           (X_STEP_BIT|Y_STEP_BIT|Z_STEP_BIT|A_STEP_BIT) // All step bits
+  #define DIRECTION_MASK      (X_DIRECTION_BIT|Y_DIRECTION_BIT|Z_DIRECTION_BIT|A_DIRECTION_BIT) // All direction bits
+ #endif
+#else
+ #if X_GANGED || X_AUTO_SQUARE
+  #define STEP_MASK           (X_STEP_BIT|X2_STEP_BIT|Y_STEP_BIT|Z_STEP_BIT) // All step bits
+  #define DIRECTION_MASK      (X_DIRECTION_BIT|X2_DIRECTION_BIT|Y_DIRECTION_BIT|Z_DIRECTION_BIT) // All direction bits
+ #elif   Y_GANGED || Y_AUTO_SQUARE
+  #define STEP_MASK           (X_STEP_BIT|Y_STEP_BIT|Y2_STEP_BIT|Z_STEP_BIT) // All step bits
+  #define DIRECTION_MASK      (X_DIRECTION_BIT|Y_DIRECTION_BIT|Y2_DIRECTION_BIT|Z_DIRECTION_BIT) // All direction bits
+ #elif   Z_GANGED || Z_AUTO_SQUARE
+  #define STEP_MASK           (X_STEP_BIT|Y_STEP_BIT|Z_STEP_BIT) // All step bits
+  #define DIRECTION_MASK      (X_DIRECTION_BIT|Y_DIRECTION_BIT|Z_DIRECTION_BIT) // All direction bits
+ #else
+  #define STEP_MASK           (X_STEP_BIT|Y_STEP_BIT|Z_STEP_BIT) // All step bits
+  #define DIRECTION_MASK      (X_DIRECTION_BIT|Y_DIRECTION_BIT|Z_DIRECTION_BIT) // All direction bits
+ #endif
+#endif
+
+// TODO: Get AutoSquaring and Ganged Axis to work with GPIO_MAP mode
```

Comment:
The second motor(s) should always use GPIO_BITBAND for setting the outputs, GPIO_MAP mode will only add complexity and likely not reduce overhead as normally only one motor is auto-squared.

Reply:
One more issue: handling of pullup for secondary limit inputs has to be added to settings_changed():

```
                case Input_LimitX:
                case Input_LimitX_Max:
                    pullup = !settings->limits.disable_pullup.x;
                    input->irq_mode = limit_fei.x ? IRQ_Mode_Falling : IRQ_Mode_Rising;
                    break;
```
etc.